src/unit.rs

```diff
@@ -378,42 +377,117 @@ fn send_prelude(unit: &Unit, stream: &mut Stream, redir: bool) -> io::Result<()>
                     _ => 0,
                 };
                 if scheme_default != 0 && scheme_default == port {
-                    write!(prelude, "Host: {}\r\n", host)?;
+                    prelude.write_header("Host", host)?;
                 } else {
-                    write!(prelude, "Host: {}:{}\r\n", host, port)?;
+                    prelude.write_header("Host", format_args!("{}:{}", host, port))?;
                 }
             }
             None => {
-                write!(prelude, "Host: {}\r\n", host)?;
+                prelude.write_header("Host", host)?;
             }
         }
     }
     if !header::has_header(&unit.headers, "user-agent") {
-        write!(prelude, "User-Agent: {}\r\n", &unit.agent.config.user_agent)?;
+        prelude.write_header("User-Agent", &unit.agent.config.user_agent)?;
     }
     if !header::has_header(&unit.headers, "accept") {
-        write!(prelude, "Accept: */*\r\n")?;
+        prelude.write_header("Accept", "*/*")?;
     }
 
     // other headers
     for header in &unit.headers {
         if !redir || !header.is_name("Authorization") {
             if let Some(v) = header.value() {
-                write!(prelude, "{}: {}\r\n", header.name(), v)?;
+                if is_header_sensitive(header) {
+                    prelude.write_sensitive_header(header.name(), v)?;
+                } else {
+                    prelude.write_header(header.name(), v)?;
+                }
```

Comment:
Wondering about the design choice here of having the `is_header_sensitive` external to `PreludeBuilder` at the same time as as the builder has `write_sensitive_header`. Not saying it's wrong, just wondering if there was a thought there?

Reply:
I think the reason I didn't include that logic inside PreludeBuilder is I wanted to re-use the `Header::is_name` function, but not all the headers we add come from an actual `Header` struct